// @shopify/cli/dist/local-4S6IAIPU.js
import {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  getThemeKitAccessDomain,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  themeToken
} from "./chunk-D5BJW27F.js";
import "./chunk-75LV6AQS.js";
import "./chunk-EG6MBBEN.js";
import "./chunk-3FBDJEGD.js";
import "./chunk-UMUTXITN.js";
import "./chunk-HMDWNGIV.js";
import "./chunk-G5R6YD27.js";
import "./chunk-G2ZZKGSV.js";
import "./chunk-UV5N2VL7.js";
import "./chunk-XE5EOEBL.js";
import "./chunk-B5EXYCV3.js";
import "./chunk-PKR7KJ6P.js";
export {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  getThemeKitAccessDomain,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  themeToken
};
